plugins/httpdispatcher.koplugin/main.lua

```diff
@@ -0,0 +1,264 @@
+    end
+    logger.dbg("HttpDispatcher: Server listening on port " .. self.port)
+end
+
+function HttpDispatcher:stop()
+    logger.dbg("HttpDispatcher: Stopping server...")
+
+    if self.http_socket then
+        self.http_socket:stop()
+        self.http_socket = nil
+    end
+    if self.http_messagequeue then
+        UIManager:removeZMQ(self.http_messagequeue)
+        self.http_messagequeue = nil
+    end
+
+    -- Plug the hole in the Kindle's firewall
+    if Device:isKindle() then
+        os.execute(string.format("%s %s %s",
+            "iptables -D INPUT -p tcp --dport", self.port,
+            "-m conntrack --ctstate NEW,ESTABLISHED -j ACCEPT"))
+        os.execute(string.format("%s %s %s",
+            "iptables -D OUTPUT -p tcp --sport", self.port,
+            "-m conntrack --ctstate ESTABLISHED -j ACCEPT"))
+    end
+
+    logger.dbg("HttpDispatcher: Server stopped.")
+end
+
+function HttpDispatcher:onRequest(host, port)
+    -- NOTE: Closure trickery because we need a reference to *this* self *inside* the callback,
+    --       which will be called as a function from another object (namely, StreamMessageQueue).
+    local this = self
+    return function(data, id_frame)
+        local request = string.match(data, "[^\n]+") or ""
+        logger.dbg("HttpDispatcher: Received request: " .. request)
+        local params_string = request:match("%u+%s+%S+%?([^%s]+)%s+%S+") -- extract params
+        logger.dbg("HttpDispatcher: Params: ", params_string)
+
+        local params_arr = {}
+
+        if params_string then
+            for params in string.gmatch(params_string, "[^&]+") do
+                -- Split each parameter string into key and value using string.match
+                local key, value = string.match(params, "(.-)=(.*)")
+                if key and value then
+                    params_arr[key] = value
+                end
+            end
+        end
+
+        local response = "HTTP/1.1 200 OK" -- start of response header
+
+        if params_arr["event"] and params_arr["data"] then
+            if params_arr["datatype"] == "number" then
+                this:dispatchEvent(params_arr["event"], tonumber(params_arr["data"]))
+            else
+                this:dispatchEvent(params_arr["event"], params_arr["data"])
+            end
+        end
+
+        response = response .. "\r\n\r\n" -- end of response header
+
+        logger.dbg("HttpDispatcher: Sending response: " .. response)
+        this.http_socket:send(response, id_frame) -- send the response back to the client
+    end
+end
+
+
+function HttpDispatcher:dispatchEvent(event, data)
+    logger.dbg("HttpDispatcher: Dispatch event " .. event .. " " .. data)
+    self.ui:handleEvent(Event:new(event, data))
+
+    if Device:isKindle() then
+        PowerD:resetT1Timeout()
+    end
+end
+
+function HttpDispatcher:addToMainMenu(menu_items)
+    menu_items.httpremote = {
+        text = _("HTTP event dispatcher"),
+        sorting_hint = ("tools"),
```

Comment:
As a note, this is good for contrib/user plugin, but if this were merged here it should use the menu order instead.